codegenerator/cli/templates/static/codegen/src/Benchmark.res

```diff
@@ -0,0 +1,207 @@
+module Data = {
+  module BlockRangeFetched = {
+    type t = {
+      stats: ChainWorker.blockRangeFetchStats,
+      chainId: int,
+      fromBlock: int,
+      toBlock: int,
+      fetchStateRegisterId: FetchState.id,
+      partitionId: int,
+      numEvents: int,
+    }
+
+    let make = (
+      ~stats,
+      ~chainId,
+      ~fromBlock,
+      ~toBlock,
+      ~fetchStateRegisterId,
+      ~partitionId,
+      ~numEvents,
+    ) => {
+      stats,
+      chainId,
+      fromBlock,
+      toBlock,
+      fetchStateRegisterId,
+      partitionId,
+      numEvents,
+    }
+
+    let schema = S.object(s => {
+      stats: s.field("stats", ChainWorker.blockRangeFetchStatsSchema),
+      chainId: s.field("chainId", S.int),
+      fromBlock: s.field("fromBlock", S.int),
+      toBlock: s.field("toBlock", S.int),
+      fetchStateRegisterId: s.field("fetchStateRegisterId", FetchState.idSchema),
+      partitionId: s.field("partitionId", S.int),
+      numEvents: s.field("numEvents", S.int),
+    })
+  }
+
+  type t = {requestStats: array<BlockRangeFetched.t>}
+  let schema = S.object(s => {
+    requestStats: s.field("requestStats", S.array(BlockRangeFetched.schema)),
+  })
+
+  let make = () => {requestStats: []}
+
+  let addBlockRangeFetched = (self, blockRangeFetched: BlockRangeFetched.t) => {
+    self.requestStats->Js.Array2.push(blockRangeFetched)->ignore
+  }
+}
+
+let data = Data.make()
+let cacheFileName = "BenchmarkCache.json"
+let cacheFilePath = NodeJsLocal.Path.join(NodeJsLocal.Path.__dirname, cacheFileName)
+
+let saveToCacheFile = data => {
+  let json = data->S.serializeToJsonStringOrRaiseWith(Data.schema)
+  NodeJsLocal.Fs.Promises.writeFile(~filepath=cacheFilePath, ~content=json)->ignore
+}
+
+let readFromCacheFile = async () => {
+  switch await NodeJsLocal.Fs.Promises.readFile(~filepath=cacheFilePath, ~encoding=Utf8) {
+  | exception _ => None
+  | content =>
+    switch content->S.parseJsonStringWith(Data.schema) {
+    | Ok(data) => Some(data)
+    | Error(e) =>
+      Logging.error(
+        "Failed to parse benchmark cache file, please delete it and rerun the benchmark",
+      )
+      e->S.Error.raise
+    }
+  }
+}
+
+let addBlockRangeFetched = (
+  ~stats,
+  ~chainId,
+  ~fromBlock,
+  ~toBlock,
+  ~fetchStateRegisterId,
+  ~partitionId,
+  ~numEvents,
+) => {
+  data->Data.addBlockRangeFetched(
+    Data.BlockRangeFetched.make(
+      ~stats,
+      ~chainId,
+      ~fromBlock,
+      ~toBlock,
+      ~fetchStateRegisterId,
+      ~partitionId,
+      ~numEvents,
+    ),
+  )
+
+  data->saveToCacheFile
+}
+
+module Summary = {
+  open Belt
+  type t = {
+    n: int,
+    mean: float,
+    stdDev: float,
+    min: float,
+    max: float,
+  }
+
+  type summaryTable = dict<t>
+
+  external logSummaryTable: summaryTable => unit = "console.table"
+
+  external arrayIntToFloat: array<int> => array<float> = "%identity"
```

Comment:
Won't  `array<int> :> array<float>` work?

Reply:
Nice! Didn't try coercing. But I think this is actually not used and was just left in the code by mistake anyway.